### DISCUSSION TITLE
Simple way to fill out a DataArray by the grid of some function parameters?

### DISCUSSION THREAD

**Discussion by max-sixty** [pydata/xarray#7498] [Q&A] [answered]
If we have some function `generate_timeseries`, like , and I want to generate a DataArray for every value of x & y.

At the moment I'm running a loop for each parameter, adding to a Dataset, and then folding each dimension into a DataArray, like:

```python

# Setup

import xarray as xr

def generate_timeseries(x, y):
    return np.random.normal(loc=x, scale=y, size=100)

N = 100
xs = [x / N for x in range(N + 1)]
ys = [y / N for y in range(N + 1)]

# Xarray code

x_ds = xr.Dataset()
for x in xs:
    y_ds = xr.Dataset()
    for y in ys:
        y_ds[y] = xr.DataArray(
            data=generate_timeseries(x=x, y=y),
            dims=["date"],
        )

    x_ds[x] = y_ds.to_array(dim="y")

da = x_ds.to_array("x")
da
```

```
Out[1]:
<xarray.DataArray (x: 101, y: 101, date: 100)>
array([[[ 0.00000000e+00,  0.00000000e+00,  0.00000000e+00, ...,
        ...,
          7.99982364e-01,  2.52652860e+00,  4.07713609e-01]]])
Coordinates:
  * y        (y) float64 0.0 0.01 0.02 0.03 0.04 ... 0.96 0.97 0.98 0.99 1.0
  * x        (x) float64 0.0 0.01 0.02 0.03 0.04 ... 0.96 0.97 0.98 0.99 1.0
Dimensions without coordinates: date

```

Has anyone found a pattern for doing this even more simply, and possibly invariant to the number of dimensions?

Thanks!

**Comment by dcherian** [comment, marked as answer]
Does xyzpy handle this case? https://xyzpy.readthedocs.io/en/latest/generate.html

**Reply by max-sixty** [reply]
This is awesome! I didn't know about this. I can't quite get it to work with a function that returns an array, but let me work on it / put an issue in. Thanks @dcherian !

**Reply by max-sixty** [reply]
Actually, if it's a DataArray, it works already!

@jcmgray  awesome library, looking forward to using it more!
